sdk/core/core-rest-pipeline/src/policies/bearerTokenChallengeAuthenticationPolicy.ts

```diff
@@ -0,0 +1,297 @@
+  /**
+   * Allows for the authentication of the main request of this policy before it's sent.
+   * The `setAuthorizationHeader` parameter received through the `ChallengeCallbackOptions`
+   * allows developers to easily assign a token to the ongoing request.
+   */
+  authorizeRequest?(options: ChallengeCallbackOptions): Promise<void>;
+  /**
+   * Allows to handle authentication challenges and to re-authorize the request.
+   * The response containing the challenge is `options.response`.
+   * The `setAuthorizationHeader` parameter received through the `ChallengeCallbackOptions`
+   * allows developers to easily assign a token to the ongoing request.
+   * If this method returns true, the underlying request will be sent once again.
+   */
+  authorizeRequestOnChallenge(options: ChallengeCallbackOptions): Promise<boolean>;
+}
+
+/**
+ * Options to configure the bearerTokenChallengeAuthenticationPolicy
+ */
+export interface BearerTokenChallengeAuthenticationPolicyOptions {
+  /**
+   * The TokenCredential implementation that can supply the bearer token.
+   */
+  credential: TokenCredential;
+  /**
+   * The scopes for which the bearer token applies.
+   */
+  scopes: string | string[];
+  /**
+   * Allows for the processing of [Continuous Access Evaluation](https://docs.microsoft.com/azure/active-directory/conditional-access/concept-continuous-access-evaluation) challenges.
+   * If provided, it must contain at least the `authorizeRequestOnChallenge` method.
+   * If provided, after a request is sent, if it has a challenge, it can be processed to re-send the original request with the relevant challenge information.
+   */
+  challengeCallbacks?: ChallengeCallbacks;
+}
+
+/**
+ * Retrieves a token from a token cache or a credential.
+ */
+export async function retrieveToken(
+  options: ChallengeCallbackOptions
+): Promise<AccessToken | undefined> {
+  const { scopes, claims, getToken, request } = options;
+
+  const getTokenOptions: GetTokenOptions = {
+    claims,
+    abortSignal: request.abortSignal,
+    tracingOptions: request.tracingOptions
+  };
+
+  return (await getToken(scopes, getTokenOptions)) || undefined;
+}
+
+/**
+ * Default authorize request
+ */
+export async function defaultAuthorizeRequest(options: ChallengeCallbackOptions): Promise<void> {
+  const accessToken = await retrieveToken(options);
+  if (!accessToken) {
+    return;
+  }
+  options.setAuthorizationHeader(accessToken.token);
+}
+
+/**
+ * We will retrieve the challenge only if the response status code was 401,
+ * and if the response contained the header "WWW-Authenticate" with a non-empty value.
+ */
+function getChallenge(response: PipelineResponse): string | undefined {
+  const challenge = response.headers.get("WWW-Authenticate");
+  if (response.status === 401 && challenge) {
+    return challenge;
+  }
+  return;
+}
+
+/**
+ * Default authorize request on challenge
+ */
+export async function defaultAuthorizeRequestOnChallenge(
```

Comment:
oh so we have default behavior now?

Reply:
.NET has default behavior in base class so we are adding default callbacks too after moving CAE into new policy.